Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -117,6 +117,75 @@ try {
     Add-Content -Path $env:GITHUB_OUTPUT -Value "BuildModes=$buildModes"
     Write-Host "BuildModes=$buildModes"
 
+    function Get-ChangedFiles($token) {
+        $headers = @{             
+            "Authorization" = "token $token"
+            "Accept" = "application/vnd.github.baptiste-preview+json"
+        }
+        $ghEvent = Get-Content $ENV:GITHUB_EVENT_PATH -encoding UTF8 | ConvertFrom-Json
+
+        if (($ENV:GITHUB_EVENT_NAME -eq "pull_request") -or ($ENV:GITHUB_EVENT_NAME -eq "pull_request_target")) {
+            $url = "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/compare/$($ghEvent.pull_request.base.sha)...$($ENV:GITHUB_SHA)"
+        } else {
+            $url = "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/compare/$($ghEvent.before)...$($ghEvent.after)"
+        }
+
+        if ($ghEvent.before -eq '0'*40) {
+            $filesChanged = @()
+        } else {
+            $response = InvokeWebRequest -Headers $headers -Uri $url | ConvertFrom-Json
+            $filesChanged = @($response.files | ForEach-Object { $_.filename })
+        }
+
+        return $filesChanged
+    }
+
+    function Get-ProjectsToBuild($settings, $projects, $baseFolder, $token) {
+        if ($settings.alwaysBuildAllProjects) {
+            Write-Host "Building all projects because alwaysBuildAllProjects is set to true"
+            return $projects
+        } elseif ($ENV:GITHUB_WORKFLOW -eq 'CI/CD') {
```

Comment:
We should have a different way to determine this than the name of the workflow.
CI/CD is the name property in the yaml file and if we at some point in time decide to change that, we will not see this.
IMO - this value should come in a parameter

Reply:
We decided to build all projects unless the workflow is triggered by pull_request or pull_request_target